dts/bindings/base/base.yaml

```diff
@@ -0,0 +1,45 @@
+---
+title: base device binding
+version: 0.1
+
+description: >
+    Binding for device
+
+properties:
+    compatible:
+        type: string
+        category: required
+        description: compatible strings
+        generation: define
+
+    reg:
+        type: array
+        description: register space
+        generation: define
+        category: optional
+
+    reg-names:
+        type: stringlist
+        description: name of each register space
+        generation: define
+        category: optional
+
+    interrupts:
+        type: array
+        category: optional
+        description: interrupts for device
+        generation: define
+
+    interrupt-names:
+        type: stringlist
+        category: optional
+        description: name of each interrupt
+        generation: define
+
+    label:
+        type: string
+        category: optional
```

Comment:
Among all the bindings in the tree, I only found these with optional labels:

- bindings/mtd/soc-nv-flash.yaml
- bindings/mtd/soc-nv-flash.yaml

Of course, I might have missed some.

Does it make sense to make `label` `category: required` here, and override it to optional in just those cases? It would save some more duplication if so.

Alternatively, perhaps those devices could just be given labels. It won't cost any code since those defines won't be used.

Reply:
I want to keep the base.yaml as all optional.

Reply:
Except for compatible, you mean? If at least one is required, why not two? Serious question.

Reply:
>  Except for compatible, you mean? If at least one is required, why not two? Serious question.

.. but given your response, this LGTM, since the other stuff is just about docstring-type changes.